src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -73,7 +73,12 @@ public object GetService(Type serviceType)
             var callCount = 0;
             return provider =>
             {
-                if (Interlocked.Increment(ref callCount) == 2)
+                // PERF: Singleton services once created will always be available in the lookup dictionary.
+                // Hence it is not needed to create an lambda expression for looking up in the dictionary.
+                // This change reduces the memory allocations during the start up because 
+                // lambda expression compilation involves expensive code access security checks for Desktop CLR.
+                var singletonCallSite = callSite as SingletonCallSite;
```

Comment:
Expression compilation allocates to an extreme degree on desktop due to CAS. https://github.com/aspnet/DependencyInjection/issues/425 This is causing a lot of heap-size growth at startup. As far as the breakdown between the impact of this change vs the change to `SingletonCallSite.Build`, @mnltejaswini will have to answer